### PR TITLE
#87623 Add New Filter for More Control Over Description Field

### PR DESCRIPTION
**Pull request by ggwicz**
_Ref:_ [C#87623](http://central.tri.be/issues/87623)

**Code review note:** 

- The big change here is the introduction of a new filter, `tribe_image_widget_instance_description`, to make the handling of the description field more customizable in response to https://github.com/moderntribe/image-widget/pull/45
- I've also tweak some of the code to bring it more in line with [our Engineering Standards](https://github.com/moderntribe/image-widget/pull/45), including adding some new docblocks to pre-existing filters.

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Whitespace found at end of line | [image-widget.php:37](https://github.com/moderntribe/image-widget/blob/35099063c004a2a3c3b1d8d6f3dcb80165d2069f/image-widget.php#L37)
:x: | Whitespace found at end of line | [image-widget.php:40](https://github.com/moderntribe/image-widget/blob/35099063c004a2a3c3b1d8d6f3dcb80165d2069f/image-widget.php#L40)
:x: | Whitespace found at end of line | [image-widget.php:113](https://github.com/moderntribe/image-widget/blob/35099063c004a2a3c3b1d8d6f3dcb80165d2069f/image-widget.php#L113)
:x: | Whitespace found at end of line | [image-widget.php:152](https://github.com/moderntribe/image-widget/blob/35099063c004a2a3c3b1d8d6f3dcb80165d2069f/image-widget.php#L152)
:x: | Expected 1 space after &quot;=&quot;; 2 found | [image-widget.php:320](https://github.com/moderntribe/image-widget/blob/35099063c004a2a3c3b1d8d6f3dcb80165d2069f/image-widget.php#L320)
:x: | Whitespace found at end of line | [image-widget.php:364](https://github.com/moderntribe/image-widget/blob/35099063c004a2a3c3b1d8d6f3dcb80165d2069f/image-widget.php#L364)
:x: | Whitespace found at end of line | [image-widget.php:554](https://github.com/moderntribe/image-widget/blob/35099063c004a2a3c3b1d8d6f3dcb80165d2069f/image-widget.php#L554)
:x: | Comma not allowed after last value in single-line array declaration | [image-widget.php:555](https://github.com/moderntribe/image-widget/blob/35099063c004a2a3c3b1d8d6f3dcb80165d2069f/image-widget.php#L555)

> via jenkins/codesniffer-phpcs

File: image-widget.php [hunk 1]
```diff
@@ -329,19 +354,32 @@ private function get_image_html( $instance, $include_link = true ) {
 		if ( ! empty( $instance['sizes'] ) ) {
 			$attr['sizes'] = $instance['sizes'];
 		}
+
+		/**
+		 * Allow filtering of the image attributes used on the front-end.
+		 *
+		 * @since TBD
```
Comment (borkweb) [inline image-widget.php]: This isn't a new filter so we can exclude this line

File: image-widget.php [hunk 2]
```diff
@@ -371,6 +409,13 @@ private function possible_image_sizes() {
 			self::CUSTOM_IMAGE_SIZE_SLUG => __( 'Custom', 'image-widget' ),
 		) );
 
+		/**
+		 * Allow filtering the image sizes available for use in the Image Widget dropdown
+		 *
+		 * @since TBD
```
Comment (borkweb) [inline image-widget.php]: This isn't a new filter so we can exclude this line

File: image-widget.php [hunk 3]
```diff
@@ -419,19 +465,26 @@ private function get_image_aspect_ratio( $instance ) {
...
 		if ( $theme_file = locate_template( array( 'image-widget/' . $template ) ) ) {
 			$file = $theme_file;
 		} else {
 			$file = 'views/' . $template;
 		}
+
+		/**
+		 * Allow filtering of currently-retrieved Image Widget template file's path.
+		 *
+		 * @since TBD
```
Comment (borkweb) [inline image-widget.php]: This isn't a new filter so we can exclude this line

**Pull request merged**